src/main/java/example/medCashFlow/services/EmployeeService.java

```diff
@@ -15,34 +19,32 @@
 @RequiredArgsConstructor
 public class EmployeeService {
 
-
     private final EmployeeRepository repository;
 
+    private final EmployeeMapper mapper;
+
     private final RoleService roleService;
 
     public Employee getEmployeeById(Long Id) {
         return repository.findById(Id).orElseThrow(EmployeeNotFoundException::new);
     }
 
+    public EmployeeResponseDTO getEmployeeResponseDTOById(Long Id) {
+        Employee employee = getEmployeeById(Id);
+        return mapper.toResponseDTO(employee);
+    }
+
     public Employee getEmployeeByEmail(String email) {
         return repository.findByEmail(email).orElseThrow(EmployeeNotFoundException::new);
     }
 
     public List<EmployeeResponseDTO> getAllEmployeesByClinicId(UUID clinicId) {
         return repository.findAllByClinicIdOrderById(clinicId).stream()
-                .map(employee -> new EmployeeResponseDTO(
-                        employee.getId(),
-                        employee.getFirstName(),
-                        employee.getLastName(),
-                        employee.getCpf(),
-                        employee.getEmail(),
-                        employee.getRole().getName(),
-                        employee.isActive()
-                )).toList();
+                .map(mapper::toResponseDTO).toList();
     }
 
     public boolean isEmployeeValid(String cpf, String email) {
-        return (isEmployeeValidByCpf(cpf) && isEmployeeValidByEmail(email));
+        return (!isEmployeeValidByCpf(cpf) || !isEmployeeValidByEmail(email));
```

Comment:
```suggestion
        return !(isEmployeeValidByCpf(cpf) || isEmployeeValidByEmail(email));
```

VALIDAR COM O GEPETO

Reply:
Não é a mesma coisa, para ser a mesma coisa deveria ser: `return (!A && !B)`, no caso acima está como `return (!A || !B). Porém eu melhorei a forma para ler a condição, deixando de mais fácil entendimento.